src/test/antoraCompletionProvider.test.ts

```diff
@@ -0,0 +1,20 @@
+import 'mocha'
+import * as vscode from 'vscode'
+import assert from 'assert'
+import AntoraCompletionProvider from '../features/antora/antoraCompletionProvider'
+import { Position } from 'vscode'
+
+let root
+
+suite('Antora CompletionsProvider', () => {
+  setup(() => {
+    root = vscode.workspace.workspaceFolders[0].uri.fsPath
+  })
+  test('Should return completion items', async () => {
+    const provider = new AntoraCompletionProvider()
+    const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/antora/multiComponents/api/modules/auth/pages/jwt/page2.adoc`))
+    const completionsItems = await provider.provideCompletionItems(file, new Position(3, 1))
+    // dumb test, WIP
+    assert.strictEqual(completionsItems.length, 3)
```

Comment:
we should probably assert that the completion items contain the right data

Reply:
yes...but for the moment I may have an issue with the insert text because all the assertions failed :(
```
 	  -    "insertText": "{xrefstyle,short@}"
      +    "insertText": "xrefstyle,short@"
```

Reply:
yes the expected value of `insertText` should be `{xrefstyle}`. The problem is that you are using:

```js
return Object.entries(attributes).map((key) => {
  // ...
})
```

`entries` will produce a list of `[key, value]` and you are using `let insertText = key.toString()` so you will get the stringify value of an array that contains the key and value:

```js
['xrefstyle', 'short@'].toString() // xrefstyle,short@
```

Instead, you should use the following:

```js
return Object.entries(attributes).map(([key, value]) => {
  const completionItem = new vscode.CompletionItem({
      label: key,
      description: value,
    }, vscode.CompletionItemKind.Text)
    let insertText = key
    insertText = prefix !== '{' ? `{${insertText}` : insertText
    insertText = suffix !== '}' ? `${insertText}}` : insertText
    completionItem.insertText = insertText
    return completionItem
})
```